flink-end-to-end-tests/test-scripts/test_sql_client.sh

```diff
@@ -61,15 +61,27 @@ for SQL_JAR in $SQL_JARS_DIR/*.jar; do
         ! [[ $EXTRACTED_FILE = "$EXTRACTED_JAR/META-INF"* ]] && \
         ! [[ $EXTRACTED_FILE = "$EXTRACTED_JAR/LICENSE"* ]] && \
         ! [[ $EXTRACTED_FILE = "$EXTRACTED_JAR/NOTICE"* ]] && \
-        ! [[ $EXTRACTED_FILE = "$EXTRACTED_JAR/org/apache/avro"* ]] ; then
+        ! [[ $EXTRACTED_FILE = "$EXTRACTED_JAR/org/apache/avro"* ]] && \
+        # Following required by amazon-kinesis-producer in flink-connector-kinesis
+        ! [[ $EXTRACTED_FILE = "$EXTRACTED_JAR/amazon-kinesis-producer-native-binaries"* ]] && \
+        ! [[ $EXTRACTED_FILE = "$EXTRACTED_JAR/cacerts"* ]] ; then
       echo "Bad file in JAR: $EXTRACTED_FILE"
       exit 1
     fi
   done
 
-  # check for proper factory
-  if [ ! -f $EXTRACTED_JAR/META-INF/services/org.apache.flink.table.factories.TableFactory ]; then
-    echo "No table factory found in JAR: $SQL_JAR"
+  # check for proper legacy table factory
+  # Kinesis connector does not support legacy Table API
+  if [[ $SQL_JAR == *"flink-sql-connector-kinesis"* ]]; then
+    echo "Skipping Legacy Table API for: $SQL_JAR"
+  elif [ ! -f $EXTRACTED_JAR/META-INF/services/org.apache.flink.table.factories.TableFactory ]; then
+    echo "No legacy table factory found in JAR: $SQL_JAR"
+    exit 1
+  fi
+
+  # check for new legacy table factory
```

Comment:
remove "new legacy" here and in the echo below, at some point this factory will not be "new"

Reply:
Done. Good spot, "new legacy" is a bit of an oxymoron :D